device/transport/amqp/test/_amqp_test.js

```diff
@@ -384,7 +395,18 @@ describe('Amqp', function () {
         transport.connect(function (err) {
           assert.isNotOk(err);
           assert(fakeBaseClient.connect.called);
-          assert.strictEqual(fakeBaseClient.connect.firstCall.args[0].sslOptions.caFile, 'ca cert');
+          assert.strictEqual(fakeBaseClient.connect.firstCall.args[0].sslOptions.ca, 'ca cert');
+          testCallback();
+        });
+      });
+
+      /*Tests_SRS_NODE_DEVICE_AMQP_06_012: [The `connect` method shall first test if the `ca` property is the name of an already existent file.  If so, it will attempt to read that file as a pem into a string value and pass the string to config object `ca` property.  Otherwise, it is assumed to be a pem string.] */
+      it('sets CA cert with contents of file if provided', function (testCallback) {
+        transport.setOptions({ ca: '.\\device\\transport\\amqp\\test\\_fake_pem.pem' });
```

Comment:
>.\\device\\transport\\amqp\\test\\_fake_pem.pem [](start = 36, length = 47)

please don't check in the file: rather create and delete it as part of the test setup/teardown.

Reply:
I thought you might say this.  Please let me know the objection.

---
In reply to: [214525511](https://github.com/Azure/azure-iot-sdk-node/pull/354#discussion_r214525511) [](ancestors = 214525511)

Reply:
Done

---
In reply to: [214986317](https://github.com/Azure/azure-iot-sdk-node/pull/354#discussion_r214986317) [](ancestors = 214986317,214525511)